docs: add sample A - 02 to header template docs [skip chromatic]

auroraVasconcelos opened the pull request:
<!-- ## Title: Please consider adding the [skip chromatic] flag to the PR title in case you dont need chromatic testing your changes. -->
## Description:
Closes https://github.com/solid-design-system/solid/issues/1448

## Definition of Reviewable:
<!-- *PR notes: Irrelevant elements should be removed.* -->
- [x] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
<!-- *If this PR includes a bug fix, an E2E test is necessary to verify the change. If the fix is purely visual, ensuring it is captured within our chromatic screenshot tests is sufficient.* -->
- [x] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

MartaPintoTeixeira commented:
After the sample titles can we have the respective description
ex: 1920px – sample/header/lg – variant A - 01

MartaPintoTeixeira commented:
The 3rd sample has different copy from FIGMA. But let me check with Coralie if we should actually translate all to english before you edit the copy here.

MartaPintoTeixeira commented:
@auroraVasconcelos I have updated the copy from DE to EN. Please change it accordingly.
https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=3502-217834&node-type=frame&t=LlHtXqbLeW5v1vXG-0

auroraVasconcelos commented:
> @auroraVasconcelos I have updated the copy from DE to EN. Please change it accordingly. https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=3502-217834&node-type=frame&t=LlHtXqbLeW5v1vXG-0

Updated

MartaPintoTeixeira commented:
When I adjust screen size (for mobile) i dont see the hamburger icon in this sample.
Can you double check?

MartaPintoTeixeira commented:
Also the hamburger menu in not working in any of the variants.

mariohamann commented on packages/components/src/templates/header.stories.ts:
```diff
@@ -36,18 +35,19 @@ export const SampleA = {
-          <sd-navigation-item><b>Über Uns</b></sd-navigation-item
-          ><sd-navigation-item><b>Märkte</b></sd-navigation-item
-          ><sd-navigation-item><b>Presseservice</b></sd-navigation-item>
+          <sd-navigation-item class="font-bold">About Us</sd-navigation-item>
+          <sd-navigation-item class="font-bold">Markets</sd-navigation-item>
+          <sd-navigation-item class="font-bold">Sustainability</sd-navigation-item>
+          <sd-navigation-item class="font-bold">Career</sd-navigation-item>
```
Please set `href`s here to make this accessible as a link. Apply everywhere.

auroraVasconcelos replied:
href="javascript:void(0)" has been added to all navigation items

mariohamann commented on packages/components/src/templates/header.stories.ts:
```diff
@@ -57,41 +57,166 @@ export const SampleA = {
+        </div>
+      </div>
+    </sd-header>
+    <sd-drawer id="sampleA__drawer" placement="end">
```
Please write constistent IDs, ideally always in kebap case.

auroraVasconcelos replied:
Updated

mariohamann commented on packages/components/src/templates/header.stories.ts:
```diff
@@ -57,41 +57,166 @@ export const SampleA = {
               ></sd-icon>
             </div>
           </sd-navigation-item>
           <sd-navigation-item>
```
Here we have a little problem: We're proposing a search icon, but do not make any suggestion what should happen, when clicking on it. Should it open a modal? Should it slide out to an text input? Should it do anything at all?

For the moment, I at least would suggest to make an alert, that at least SOMETHING happens. Or you could make it a link, indicating that we would switch to an external search page.

karlbaumhauer replied:
@mariohamann isn't this the case for ALL buttons/items in the header? If think we at least add links to each of the item or (in case of a button) trigger an alert on the page.

auroraVasconcelos replied:
Hey @karlbaumhauer and @mariohamann,
I believe it's unnecessary to make everything clickable, as this seems to be a template, not a real case example. IMO making every element interactive would add unnecessary complexity to the code, making it harder to clean up and reuse.
For now, I've made only the search icon clickable in each sample. It opens an alert that says, "This could open a search bar.

mariohamann commented on packages/components/src/templates/header.stories.ts:
```diff
@@ -57,41 +57,166 @@ export const SampleA = {
+        </sd-navigation-item>
+      </nav>
+    </sd-drawer>
+    <script>
+      // This line is using querySelectorAll due to hidden elements from Storybook configs.
+      // In a regular environment, use: var drawer = document.getElementById('sampleA') or similar
+      var drawerSampleA = document.querySelectorAll('#sampleA__drawer')[1];
+      console.log(document.querySelectorAll('#sampleA__drawer')[1]);
+      function openDrawerSampleA() {
+        drawerSampleA.show();
+      }
+    </script>
```
Make this a type=module and do not use `var` but const instead.

Remove the console.log.

Why can't you give it a proper ID that is unique? Please adapt, that we can use getElementByID.

Adapt for the other example.

auroraVasconcelos replied:
- Var to const updated
- Console.log removed
- type=module is currently breaking, wasn't changed due to it

> Why can't you give it a proper ID that is unique? Please adapt, that we can use getElementByID.
Due to storybook duplication of the stories, there are two ids that are not possible to be prevented. Bug ticket for it: https://github.com/solid-design-system/solid/issues/1566

auroraVasconcelos commented:
> When I adjust screen size (for mobile) i dont see the hamburger icon in this sample. Can you double check?



> Also the hamburger menu in not working in any of the variants.

Updated

auroraVasconcelos commented on packages/components/src/templates/header.stories.ts:
```diff
@@ -121,38 +245,59 @@ export const SampleB = {
+      function openDrawerSampleB() {
+        const drawerSampleB = document.getElementById('sample-b-drawer');
+        drawerSampleB.show();
+        0;
```
Updated

MartaPintoTeixeira commented:
When the drawer is open in the hamburger format there is a bug in the sd-navigation items. Create a fix ticket (https://github.com/solid-design-system/solid/issues/1575) to fix the alignment of this bug.

MartaPintoTeixeira reviewed (approved):
Approved. There is a bug ticket to fix the alignment of the navigation item where there is icon + link (horizontal)
